solana/program_v2/programs/gateway_v2/src/instructions/admin/update_network.rs

```diff
@@ -69,12 +67,14 @@ pub struct UpdateNetworkAccount<'info> {
     network.gatekeepers.len(),
     network.supported_tokens.len() + data.supported_tokens.add.len() - data.supported_tokens.remove.len()
     ),
-    realloc::payer = authority,
+    realloc::payer = payer,
     realloc::zero = false,
+    constraint = network.can_access(&authority, NetworkKeyFlags::SET_EXPIRE_TIME) @ NetworkErrors::InsufficientAccessExpiry,
+    constraint = network.can_access(&authority, NetworkKeyFlags::AUTH) @ NetworkErrors::InsufficientAccessAuthKeys,
     )]
```

Comment:
These constraints aren't correct... e.g. checking if the authority has access to set the expire time should only happen if the expire time is changed

Reply:
Should we maybe change the permissions to JUST reflect the instructions? then we don't need these fine-grained checks that might be error prone to implement.

But of course there is a downside of less flexibility.

Reply:
I feel like big update function are always a bit tricky. Not sure it makes sense for some of the access control to be `AUTH` and one being `SET_EXPIRE_TIME`.

Reply:
@rado0x54 @william-brooks  what should we update these constraints to?

Reply:
Currently it requires both `SET_EXPIRE_TIME` and `AUTH` to be set for the key...

it should be:
* (if expiry time is being changed) && (has_access(SET_EXPIRY_TIME) )
* (if auth keys are being changed) && (has_access(AUTH))

We are actually also missing checks for FEATURES, FEES, TOKENS...

Reply:
Okay thanks!